### DISCUSSION TITLE
Use inlay hints to display type after calling `reveal_type()`

### DISCUSSION THREAD

**Discussion by diepala** [microsoft/pylance-release#3315] [Enhancement]
Calling `reveal_type()` allows seeing the type of a variable, but it is needed to hover over the variable name to see the type. This makes using the `reveal_type()` function not very practical, as I can already see the type of a variable by hovering over it without using the reveal type function.

It would be better if, when calling the reveal type function, an inlay hint appears revealing the type of the variable.
The same could be done with the function `reveal_locals()`.

**Comment by erictraut** [comment]
Thanks for the suggestion.

As you said, you can already see the type of a symbol (an identifier) by hovering over it. The `reveal_type` call allows you to reveal the type of an _expression_, which can be a single identifier or something more complex. So I see the appeal of adding support for inlay hints for `reveal_type`.

In Python 3.11, the `reveal_type` function was standardized and added to the `typing` module (and `typing_extensions` module) and can now be left in your code at runtime. The `reveal_locals` function was not standardized because it is rarely used. So if we add inlay hint support, I would recommend doing it for `reveal_type` but not `reveal_locals`.
